privacyidea/lib/auditmodules/base.py

```diff
@@ -88,32 +89,11 @@ def __init__(self, config=None):
         :return:
         """
         self.name = "AuditBase"
-        self.audit_data = {}
+        self.audit_data = {'startdate': datetime.datetime.now()}
```

Comment:
Will look into this and lets discuss.

Reply:
I think there are not that many steps *Before* creating the audit object, as the audit object gathers all data during the request. If we would wanted to do this that early (we also would actually have the same minimal offset without any real interaction, since no configuration is yet read, no communication to the database...) we would have to save the time in ``g`` and then pass it to the audit. I am not sure if we would overcomplicate things or if in ``before_app_request`` we have all necessary information.

However, we can move the creation of the audit object [here](https://github.com/privacyidea/privacyidea/blob/8a076b7b56b35c298c25b89d1fb044b3eaedba63/privacyidea/api/before_after.py#L160) to the beginning of the function ``def before_request``.

Reply:
done in 9381eac1